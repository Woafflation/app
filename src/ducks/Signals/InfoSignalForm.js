import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Formik, Form, Field } from 'formik'
import { Input, Button } from '@santiment-network/ui'
import styles from './TriggerForm.module.scss'

const InfoSignalForm = ({
  title = 'Signal-' + new Date().toISOString(),
  description = '',
  isEdit = false,
  onInfoSignalSubmit,
  onBack
}) => {
  return (
    <Fragment>
      <Formik
        initialValues={{ title, description }}
        isInitialValid
        validate={values => {
          let errors = {}
          if (!values.title) {
            errors.title = 'Required'
          } else if (values.title.length < 3) {
            errors.title = 'Title has to be longer than 2 characters'
          }
          return errors
        }}
        onSubmit={(values, { setSubmitting }) => {
          onInfoSignalSubmit(values)
        }}
      >
        {({
          values,
          errors,
          touched,
          isSubmitting,
          handleChange,
          handleBlur,
          isValid,
          ...rest
        }) => (
          <Form>
            <div className={styles.Field}>
              <label>Name of the signal</label>
              <Field
                id='title'
                name='title'
                type='text'
                placeholder='Name of signal'
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.title}
                component={Input}
              />
              {errors.title && touched.title && errors.title}
            </div>
            <div className={styles.Field}>
              <label>Description</label>
              <Field
                id='description'
                component={Input}
                type='text'
                name='description'
                value={values.description}
                onChange={handleChange}
                onBlur={handleBlur}
                placeholder='Description of the signal'
              />
            </div>
            <div className={styles.controls}>
              <Button
                type='button'
                variant={'flat'}
                accent='normal'
                border
                onClick={onBack}
              >
                Back
              </Button>
              <Button
                type='submit'
                disabled={!isValid || isSubmitting}
                isActive={isValid && !isSubmitting}
                variant={'fill'}
                accent='positive'
              >
                {isEdit ? 'Update' : 'Create'}
              </Button>
            </div>
          </Form>
        )}
      </Formik>
    </Fragment>
  )
}

InfoSignalForm.propTypes = {
  onInfoSignalSubmit: PropTypes.func.isRequired
}

export default InfoSignalForm
