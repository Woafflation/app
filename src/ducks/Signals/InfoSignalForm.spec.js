/* eslint-env jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import InfoSignalForm from './InfoSignalForm'

describe('InfoSignalForm', () => {
  it('render correctly text component', () => {
    const constantDate = new Date('2017-06-13T04:41:20')

    /* eslint no-global-assign:off */
    Date = class extends Date {
      constructor () {
        return constantDate
      }
    }
    const wrapper = shallow(
      <InfoSignalForm onGoBack={() => {}} onInfoSignalSubmit={() => {}} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should be disabled, if title is empty', () => {
    const mockCb = jest.fn()
    const wrapper = mount(<InfoSignalForm onInfoSignalSubmit={mockCb} />)
    const inputTitle = wrapper.find('#title').at(1)
    inputTitle.simulate('change', {
      target: {
        value: '',
        name: 'title'
      }
    })
    console.log(wrapper.find('input'))
    expect(
      wrapper
        .find('[type="submit"]')
        .at(1)
        .prop('disabled')
    ).toBe(true)
  })
})
