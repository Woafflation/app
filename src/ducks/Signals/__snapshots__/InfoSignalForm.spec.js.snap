// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoSignalForm render correctly text component 1`] = `
<Fragment>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "description": "",
        "title": "Signal-2017-06-12T21:41:20.000Z",
      }
    }
    isInitialValid={false}
    onChange={[Function]}
    onSubmit={[Function]}
    validate={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
  >
    <Component />
  </Formik>
</Fragment>
`;
